File: code_wars/kyu_7/012-Sum_of_array_singles.py
###### Solution 1
def repeats(arr: list) -> int:
    result = 0

    for num in arr:
        if arr.count(num) == 1:
            result += num

    return result


###### Solution 2
def repeats(arr: list) -> int:
    return sum(num for num in arr if arr.count(num) == 1)


###### Solution 3 (Only works if the max duplication = 2)
def repeats(arr: list) -> int:
    return 2 * sum(set(arr)) - sum(arr)


###### Video Link:
###### Problem Link: https://www.codewars.com/kata/59f11118a5e129e591000134/train/javascript

File: code_wars/kyu_7/012-Sum_of_array_singles.js
// ###### Solution 1
function repeats(arr) {
  let result = 0;

  for (let i = 0; i < arr.length; i++) {
    if (arr.filter((ele) => ele === arr[i]).length === 1) result += arr[i];
  }

  return result;
}

// ###### Solution 2
function repeats(arr) {
  return arr.reduce(
    (acc, cur) =>
      arr.filter((ele) => ele === cur).length === 1 ? acc + cur : acc,
    0
  );
}

// ###### Solution 3
function repeats(arr) {
  return (
    2 * Array.from(new Set(arr)).reduce((a, c) => a + c) -
    arr.reduce((a, c) => a + c)
  );
}

// ###### Video Link:
// ###### Problem Link: https://www.codewars.com/kata/59f11118a5e129e591000134/train/javascript
